### PR TITLE
Deprecating old typing settings to standardize

### PR DESCRIPTION
**Pull request by Tang8330**
Standardizing everyone to use the new format. When publishing this, we'll publish a separate version.

### CODE REVIEW

File: lib/config/config.go [hunk 1]
```diff
@@ -95,21 +95,7 @@ type SharedDestinationConfig struct {
...
-func (stc SharedTransferConfig) ToTypingSettings() typing.Settings {
-	if stc.TypingSettings != nil {
-		return *stc.TypingSettings
-	}
-
-	return typing.Settings{
-		AdditionalDateFormats:       stc.AdditionalDateFormats,
-		CreateAllColumnsIfAvailable: stc.CreateAllColumnsIfAvailable,
-	}
+	TypingSettings typing.Settings `yaml:"typingSettings"`
```
Comment (Tang8330) [inline lib/config/config.go:98]: Got rid of the pointer so it'll return an empty struct.

File: lib/config/config_test.go [hunk 2]
```diff
@@ -20,22 +20,17 @@ func TestSharedTransferConfig(t *testing.T) {
...
 		assert.NoError(t, err)
 
-		assert.Equal(t, "yyyy-MM-dd", sharedTransferCfg.AdditionalDateFormats[0])
-		assert.True(t, sharedTransferCfg.CreateAllColumnsIfAvailable)
-
 		assert.True(t, sharedTransferCfg.TypingSettings.CreateAllColumnsIfAvailable)
 		assert.Equal(t, "yyyy-MM-dd1", sharedTransferCfg.TypingSettings.AdditionalDateFormats[0])
-		assert.True(t, sharedTransferCfg.ToTypingSettings().CreateAllColumnsIfAvailable)
-		assert.Equal(t, "yyyy-MM-dd1", sharedTransferCfg.ToTypingSettings().AdditionalDateFormats[0])
+		assert.True(t, sharedTransferCfg.TypingSettings.CreateAllColumnsIfAvailable)
```
Comment (nathan-artie) [inline lib/config/config_test.go:32]: These two assertions are now the same as the two on lines 35-36.

File: lib/config/config_test.go [hunk 3]
```diff
@@ -389,11 +389,12 @@ reporting:
...
-	assert.Equal(t, config.Kafka.Username, username)
-	assert.Equal(t, config.Kafka.BootstrapServer, bootstrapServer)
-	assert.Equal(t, config.Kafka.GroupID, groupID)
-	assert.Equal(t, config.Kafka.Password, password)
+	assert.Equal(t, true, config.Kafka.EnableAWSMSKIAM)
+	assert.Equal(t, username, config.Kafka.Username)
+	assert.Equal(t, bootstrapServer, config.Kafka.BootstrapServer)
+	assert.Equal(t, groupID, config.Kafka.GroupID)
+	assert.Equal(t, password, config.Kafka.Password)
+	assert.Equal(t, true, config.SharedTransferConfig.TypingSettings.CreateAllColumnsIfAvailable)
```
Comment (nathan-artie) [inline lib/config/config_test.go:397]: `assert.True(`

File: lib/config/config_test.go [hunk 4]
```diff
@@ -389,11 +389,12 @@ reporting:
 	assert.Nil(t, err)
 	assert.NotNil(t, config)
 
-	assert.Equal(t, config.Kafka.EnableAWSMSKIAM, true)
-	assert.Equal(t, config.Kafka.Username, username)
-	assert.Equal(t, config.Kafka.BootstrapServer, bootstrapServer)
-	assert.Equal(t, config.Kafka.GroupID, groupID)
-	assert.Equal(t, config.Kafka.Password, password)
+	assert.Equal(t, true, config.Kafka.EnableAWSMSKIAM)
```
Comment (nathan-artie) [inline lib/config/config_test.go:392]: `assert.True(`

**Review by nathan-artie — APPROVED** [review]

**Pull request merged**